client/src/client.rs

```diff
@@ -460,7 +459,9 @@ impl Client {
 
         let mut tx_builder = match instructions.into() {
             Executable::Instructions(instructions) => tx_builder.with_instructions(instructions),
-            Executable::Wasm(wasm) => tx_builder.with_wasm(wasm),
+            Executable::SmartContract(smart_contract) => {
```

Comment:
Imo what corresponds to smartcontracts is triggers in Iroha, and the execution part of a trigger (and a transaction) can be Wasm or ISIs. So I don't think "smartcontract" is a good candidate to rename "wasm"

Reply:
While working on #4850, I noticed that Iroha has both triggers and smart contracts, at least we have two separate macros for defining the entry point. Not sure what the difference is.

Reply:
Ok, so smart contracts only have an owner, but triggers also specify the trigger id, owner and event. What happens when a trigger wasm is executed as a smart contract, and vice versa?

Reply:
> While working on #4850, I noticed that Iroha has both triggers and smart contracts, at least we have two separate macros for defining the entry point. Not sure what the difference is.

how I see it.

1. `Transaction::executable` is either a `Vec<Instruction>` or `SmartContract`
You can execute an executable with `client.submit`. This executes the instructions or the smart contract immediately on submition
2. `Action::executable` is either a `Vec<Instruction>` or `SmartContract`
You have to first register a trigger and then invoke it with `InstructionBox::Execute`. The trigger can consist of  `Vec<InstructinoBox`> or a be a smart contract

Reply:
> Ok, so smart contracts only have an owner, but triggers also specify the trigger id, owner and event

no, smart contracts don't have an owner. The transaction they are part of has an owner

Reply:
I mean a smart contract entrypoint has an owner field, while a trigger entrypoint has that plus additional fields